pkg/operator/api/api.go

```diff
@@ -182,6 +187,26 @@ func (a *apiServer) GetConfiguration(ctx context.Context, in *operatorv1pb.GetCo
 
 // ListComponents returns a list of Dapr components.
 func (a *apiServer) ListComponents(ctx context.Context, in *operatorv1pb.ListComponentsRequest) (*operatorv1pb.ListComponentResponse, error) {
+	// by default assume that components are not getting loaded for control plane service
+	controlPlaneServiceReq := false
+	spiffeID, ok, err := spiffe.FromGRPCContext(ctx)
+	if err != nil {
+		log.Debugf("failed to get SPIFFE ID from gRPC connection context: %v", err)
+		return nil, err
+	}
+	if !ok {
+		// Apply the default action
+		log.Debugf("Error while reading spiffe id from client cert. applying default global policy action")
+	}
+
+	if in.GetNamespace() == security.CurrentNamespace() {
+		for _, service := range a.additionalCPServices {
```

Comment:
I think here we need to list all control plane services. The "additional" ones are ...additional. But we should have the built-in ones too, such as sentry & placement (not that it's needed today, but for completness)

Reply:
Had not added them - but as you said, for the sake of completeness - added them and as per another comment, shifted to utils, to be used by operator and x509source.